functests/2_performance_update/kubelet.go

```diff
@@ -0,0 +1,170 @@
+package __performance_update
+
+import (
+	"context"
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	performancev2 "github.com/openshift-kni/performance-addon-operators/api/v2"
+	testutils "github.com/openshift-kni/performance-addon-operators/functests/utils"
+	testclient "github.com/openshift-kni/performance-addon-operators/functests/utils/client"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/discovery"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/mcps"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/nodes"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/profiles"
+	machineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strconv"
+	"strings"
+)
+
+var _ = Describe("[ref_id: 45487][pao]additional kubelet arguments", func() {
+	var profile *performancev2.PerformanceProfile
+	var workerRTNodes []corev1.Node
+	var performanceMCP string
+
+	testutils.BeforeAll(func() {
+		workerRTNodes, err := nodes.GetByLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+
+		workerRTNodes, err = nodes.MatchingOptionalSelector(workerRTNodes)
+		Expect(err).ToNot(HaveOccurred())
+
+		profile, err = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+
+		performanceMCP, err = mcps.GetByProfile(profile)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Verify that worker and performance MCP have updated state equals to true
+		for _, mcpName := range []string{testutils.RoleWorker, performanceMCP} {
+			mcps.WaitForCondition(mcpName, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+		}
+
+	})
+	BeforeEach(func() {
+		if discovery.Enabled() && testutils.ProfileNotFound {
+			Skip("Discovery mode enabled, performance profile not found")
+		}
+	})
+	Context("Additional kubelet arguments", func() {
+		It("[test_id:45488]Test performance profile annotation for changing multiple kubelet settings", func() {
+			sysctlAnnotations := map[string]string{
+				"kubeletconfig.experimental": "{\"allowedUnsafeSysctls\":[\"net.core.somaxconn\",\"kernel.msg*\"],\"systemReserved\":{\"memory\":\"300Mi\"},\"kubeReserved\":{\"memory\":\"768Mi\"},\"imageMinimumGCAge\":\"3m\"}",
+			}
+			profile.SetAnnotations(sysctlAnnotations)
+			By("Applying changes in performance profile and waiting until mcp will start updating")
+			profiles.UpdateWithRetry(profile)
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+			By("Waiting when mcp finishes updates")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
```

Comment:
same check for the error.

Reply:
@mrniranjan   mcps.WaitForCondition  is a function that returns an error in case it fails.
I suggest follow @cynepco3hahue 's suggestion here in all of the function occurrences and either use the Nil check as above or have:
```
err := mcps.WaitForCondition
Expect(err).ToNot(HaveOccurred())
```
I think we can check for the error only for `Updated` state wait since we might miss `Updating` state if its too "fast"
and in the test cases here they seem to be sequential.

Reply:
@yanirq  I tried but i could not make it to work as Based on [mcp.WaitForCondition](https://github.com/openshift-kni/performance-addon-operators/blob/v0.0.41001/functests/utils/mcps/mcps.go#L159) Code, it doesn't return anything.